### ISSUE TITLE
Building ord for Windows from source using syntax in readme results in bad build

### ISSUE THREAD

**Issue by cryptoni9n** [ordinals/ord#3170]
**tl;dr:** Should we add a checkout command to the instructions on how to build ord from source on https://github.com/ordinals/ord?tab=readme-ov-file#building?

Hello all - please pardon any etiquette mistakes in this Issue report, it is my first and I'm not yet experienced with Github.

I built ord 0.15.0 for Windows from source using the Github clone commands found in the readme, here: https://github.com/ordinals/ord?tab=readme-ov-file#building

I then built the index with sats using 0.15.0 and everything seemed to be working fine.  The server would start and was accessible on the localmachine, the wallet I had started Bitcoin Core with would return a balance, ord --version returns 'ord 0.15.0', etc.  My issue is when I try to restore a non-ord wallet into ord, I receive the following error: 

> error: JSON-RPC error: transport error: Couldn't connect to host: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (os error 10060)
>    0: std::backtrace_rs::backtrace::dbghelp::trace
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
>    1: std::backtrace_rs::backtrace::trace_unsynchronized
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
>    2: std::backtrace::Backtrace::create
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\backtrace.rs:332
>    3: std::backtrace::Backtrace::capture
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\backtrace.rs:298
>    4: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
>    5: ord::wallet::Wallet::exists
>    6: ord::subcommand::wallet::restore::Restore::run
>    7: ord::subcommand::wallet::WalletCommand::run
>    8: ord::subcommand::Subcommand::run
>    9: ord::main
>   10: __ImageBase
>   11: __ImageBase
>   12: std::rt::lang_start_internal::closure$2
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\rt.rs:148
>   13: std::panicking::try::do_call
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\panicking.rs:500
>   14: std::panicking::try
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\panicking.rs:464
>   15: std::panic::catch_unwind
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\panic.rs:142
>   16: std::rt::lang_start_internal
>              at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\rt.rs:148
>   17: main
>   18: invoke_main
>              at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
>   19: __scrt_common_main_seh
>              at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
>   20: BaseThreadInitThunk
>   21: RtlUserThreadStart
> 

I sought help on the Ordicord and Discord mod **ETS** recommended I try downloading a prebuilt version of ord instead of using my own.  I did so and attempted to use the index that I had built with my initial build of ord 0.15.0, but received this error:

> error: index at `F:\Bitcoin\ord_index\0.15.0\index.redb` appears to have been built with a newer, incompatible version of ord, consider updating ord: index schema 17, ord schema 16
>    0: <unknown>
>    1: <unknown>
>    2: <unknown>
>    3: <unknown>
>    4: <unknown>
>    5: <unknown>
>    6: <unknown>
>    7: <unknown>
>    8: <unknown>
>    9: <unknown>
>   10: <unknown>
>   11: BaseThreadInitThunk
>   12: RtlUserThreadStart

Discord mod **ETS** mentioned that this seemed to prove what he thought the issue may be - building from source using the syntax from the readme doesn't actually build the release as it was released, but instead builds the release with every single change in the source up until that point.  This isn't always a functioning build and can result in some errors for some functionality depending on the changes in the build at the time - at least, that is the issue as I understand it.

This was discussed in the Ordicord at the following links: https://discord.com/channels/987504378242007100/1069465367988142110/1209130700738138142 and https://discord.com/channels/987504378242007100/987504378749538366/1207049658002837514

According to Discord mod **Greg**, there should be a "git checkout 0.15.0 before the cargo command".

I think the fix here is just to update the readme directions.  I would love to be involved in submitting the fix or testing it or following this through to the end because I need experience in Github and I really want to contribute in any way possible to the project.

Please let me know if there is any further information that I can provide.
-cryptoni9n

**Comment by cryptoni9n** [comment]
**Update**: attempting to test the proposed fix with 0.16.0-rc1 by submitting the checkout request prior to building the release.  The following advice is presented:
```

PS C:\Users\xxx\ord16\ord> git checkout 0.16.0-rc1                      
Note: switching to '0.16.0-rc1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at e5f48e6b Release 0.16.0-rc1 (#3185)
```

I'll provide more updates as this build and index complete in order to determine if this has resolved my original issue.

**Comment by cryptoni9n** [comment]
I was able to successfully build ord from source using the checkout command, so it works - but I'm not 100% sure if it is necessary.  Deferring to higher minds.

**Comment by raphjaph** [comment]
If you want to build from `master` you just clone the repo and do `cargo build --release`
If you want to build from a release you just do `git checkout 0.14.1` and `cargo build --release`

This seems to work now, so I'll close the issue. Please reopen if not

**Comment by so7ow** [comment]
> If you want to build from master you just clone the repo and do cargo build --release
> If you want to build from a release you just do git checkout 0.14.1 and cargo build --release

I think the point is that the build instructions @ https://github.com/ordinals/ord?tab=readme-ov-file#building lead users to build incorrectly as they don't include a call to a version checkout.

**Comment by raphjaph** [comment]
Build instructions on repos always assume building from `master`. We could add a small line that says how to checkout a specific version with `git` but the default should be `master`

**Comment by raphjaph** [comment]
https://github.com/ordinals/ord/pull/3304
